src/unix/bsd/netbsdlike/netbsd/riscv64.rs

```diff
@@ -3,11 +3,94 @@ use PT_FIRSTMACH;
 pub type c_long = i64;
 pub type c_ulong = u64;
 pub type c_char = u8;
+pub type __greg_t = u64;
 pub type __cpu_simple_lock_nv_t = ::c_int;
 
+s! {
+    pub struct mcontext_t {
+        pub __gregs: [::greg_t; 32],
+        pub __fregs: [__c_anonymous__fpreg; 33],
+        __spare: [::__greg_t; 7],
+    }
+}
+
+s_no_extra_traits! {
+    pub union __c_anonymous__fpreg {
+        pub u_u64: u64,
+        pub u_d: ::c_double,
+    }
+}
+
+cfg_if! {
+    if #[cfg(feature = "extra_traits")] {
+        impl PartialEq for __c_anonymous__fpreg {
+            fn eq(&self, other: &__c_anonymous__fpreg) -> bool {
+                unsafe {
+                self.u_u64 == other.u_u64
+                    || self.u_ud == other.u_ud
+                }
+            }
+        }
+        impl Eq for __c_anonymous__fpreg {}
```

Comment:
These should probably just be dropped, it would be weird to use float equality to compare integers.